luci-app-wol: replace fs.stat/exec with safe RPC backend, fix ACLs

mdevolde opened the pull request:
- [x] This PR is not from my *main* or *master* branch :poop:, but a *separate* branch :white_check_mark:
- [x] Each commit has a valid :black_nib: `Signed-off-by: <my@email.address>` row (via `git commit --signoff`)
- [x] Each commit and PR title has a valid :memo: `<package name>: title` first line subject for packages
- [x] Incremented :up: any `PKG_VERSION` in the Makefile
- [x] Tested on: aarch64, OpenWrt 24.10.4, Opera One(version : 124.0.5705.42) :white_check_mark: 
- [x] \( Preferred ) Mention: @mdevolde
- [x] \( Preferred ) Screenshot or mp4 of changes:

<img width="977" height="391" alt="518189561-3d2b61e8-5241-49e5-9058-54c46eb06236" src="https://github.com/user-attachments/assets/f45465b8-fa11-4265-88ff-e5a4c7381036" />

<img width="984" height="391" alt="518201649-0ff4c5e8-3eb5-43c6-90fc-0a18e9200d47" src="https://github.com/user-attachments/assets/216288ad-6c71-48b4-a0da-2cf6ec31b14d" />


Screenshots of functional access to `luci-app-wol` for a restricted user (restricted to these permissions : read `unauthenticated`, read `luci-base`, read `luci-app-wol`, write `luci-app-wol`).

- [x] Description:

In its current state, if we create a restricted user as mentioned above, access to `luci-app-wol` does not work because it cannot verify that the two executables `/usr/bin/etherwake` and `/usr/bin/wol` exist via `fs.stat`.
Furthermore, it cannot be executed via `fs.exec` either, as ACLs are missing in `luci-app-wol.json` (`ubus file`).
Finally, an ACL concerning the use of `DeviceSelect` is also missing.

Instead of adding broad `ubus file.*` ACLs, a dedicated `luci.wol` backend has been added, providing two RPC methods:
- `stat` to verify that the two executables exist and are executable
- `exec` to execute one of these two executables, with parameters.

The missing ACL for using `DeviceSelect` has also been added (`getNetworkDevices`).

Adding this backend and the appropriate ACLs makes the `luci-app-wol` app functional, even for a restricted user, by assigning only the permissions necessary for it to work properly.

Signed-off-by: Martin Devolder <martin.devolder2@gmail.com>

systemcrash commented on applications/luci-app-wol/root/usr/libexec/rpcd/luci.wol:
```diff
@@ -0,0 +1,90 @@
+#!/bin/sh
```
recommend this be written as ucode, then you avoid manual json handling pitfalls, and just build objects directly.

mdevolde commented on applications/luci-app-wol/htdocs/luci-static/resources/view/wol.js:
```diff
@@ -19,16 +33,16 @@ return view.extend({
 		]);
 	},
 
 	render: function(data) {
```
fixed

mdevolde commented:
Rebased and force-pushed with additional updates.
Adapted `luci-app-wol` to recent versions of OpenWrt where the `wol` binary is no longer provided. The code now switches to `wakeonlan`, which is the current replacement for the `wol` package. Updated the paths, existence tests, executable selection, and corrected the call by removing the irrelevant -v option.
Also corrected an incorrect reference to `data.executable` by using `data.wol.executable`.

mdevolde commented:
@systemcrash Done, each line of the commit title and description is now less than 80 characters long.

mdevolde commented on applications/luci-app-wol/root/usr/share/rpcd/acl.d/luci-app-wol.json:
```diff
@@ -3,14 +3,14 @@
 			"ubus": {
-				"luci-rpc": [ "getHostHints" ]
+				"luci.wol": [ "stat" ],
+				"luci-rpc": [ "getHostHints", "getNetworkDevices" ]
```
Yep, it's used internally by `widget.DeviceSelect`

mdevolde replied:
In `modules\luci-base\htdocs\luci-static\resources\tools\widgets.js`, at line 564, `DeviceSelect` use `network.getDevices()`.
In `modules\luci-base\htdocs\luci-static\resources\network.js`, at line 1159, `getDevices` use `initNetworkState()`.
In `modules\luci-base\htdocs\luci-static\resources\network.js`, at line 359, `initNetworkState` use `callLuciNetworkDevices()` :
```js
var callLuciNetworkDevices = rpc.declare({
	object: 'luci-rpc',
	method: 'getNetworkDevices',
	expect: { '': {} }
});
```

Without this permission, `DeviceSelect` cannot retrieve existing devices, and users who choose to use `etherwake` cannot select one of the existing devices in the web interface of `luci-app-wol`.

This is for the first field here:
<img width="977" height="391" alt="image" src="https://github.com/user-attachments/assets/ef149407-a935-4824-8d18-1b1ecd030e52" />

systemcrash replied:
Good catch!

systemcrash commented:
Merged. Thanks @mdevolde !

systemcrash commented:
I usually give it a week and see how it fares in master - with any follow-up fixes coming shortly thereafter. It will land in the next release, and weeklies.
